### PR TITLE
Prevent the Default Template from getting deleted

### PR DESCRIPTION
**Pull request by grivero**
[W-038607](https://foundation.lightning.force.com/lightning/r/agf__ADM_Work__c/a2x1E000001HqfXQAS/view): Prevent the Default Template from getting deleted

# Critical Changes

# Changes
- add an additional validation onBeforeDelete for Form_Template__c in `BGE_FormTemplate_TDTM`
- add new method in `UTIL_CustomSettings_API` to get Gift_Entry_Settings__c instance

# Issues Closed

# Community Ideas Delivered

# New Metadata

# Deleted Metadata

### CODE REVIEW

**Comment by grivero** [comment]
**lurch: attach W-038607

**Comment by LurchTheButler** [comment]
Tracking <a href="https://foundation.my.salesforce.com/a2x1E000001HqfXQAS" target="_blank">W-038607</a>

File: src/classes/BGE_FormTemplate_TDTM.cls [hunk 1]
```diff
@@ -46,39 +46,56 @@ public with sharing class BGE_FormTemplate_TDTM extends TDTM_Runnable{
...
-     * @param triggerAction which trigger event (BeforeInsert, AfterInsert, etc.)
      *******************************************************************************************************/
-    private void validateFormTemplateDeletion (List<SObject> newlist, List<SObject> oldlist,
-            TDTM_Runnable.Action triggerAction){
+    private void validateFormTemplateDeletion ( List<SObject> oldlist ){
 
-        if (triggerAction == TDTM_Runnable.Action.BeforeDelete) {
-            List<Form_Template__c> formTemplates = (List<Form_Template__c>) oldlist;
+        // aux vars
+        List<Form_Template__c>      formTemplates           = (List<Form_Template__c>) oldlist;
```
Comment (vmartinov-sforg) [inline src/classes/BGE_FormTemplate_TDTM.cls]: Let's remove the extra spaces around the '=', to be consistent with the npsp code formatting.

**Comment by grivero** [comment]
On it Vladimir

File: src/classes/BGE_FormTemplate_TDTM.cls (hunk 1 above)
Comment (grivero) [reply src/classes/BGE_FormTemplate_TDTM.cls]: I was trying to improve variable legibility (check img below) but i'm going to do as you said and keep consistency with rest of the code.

Screen Shot 2020-01-24 at 10 38 01 AM

File: src/classes/FORM_ServiceGiftEntry_TEST.cls [hunk 2]
```diff
@@ -249,6 +249,53 @@ public with sharing class FORM_ServiceGiftEntry_TEST {
...
     }
 
+    /**************************************************************************************************
+    * @description Confirms that a form template used as default on Gift Entry settings cannot be deleted.
+    */
+    @IsTest
+    static void shouldPreventDeletionOfTemplateReferencedByGiftEntry() {
+
+        // aux vars
+        String sTemplateName    = 'fooName';
```
[the remark this answers is not in the document]
Comment (grivero) [reply src/classes/FORM_ServiceGiftEntry_TEST.cls]: 👍

File: src/classes/FORM_ServiceGiftEntry_TEST.cls [hunk 3]
```diff
@@ -249,6 +249,53 @@ public with sharing class FORM_ServiceGiftEntry_TEST {
...
+        AuraHandledException failure;
+
+        // test data
+        FORM_Template template  = createSampleTemplate();
+        template.name           = sTemplateName;
```
[the remark this answers is not in the document]
Comment (grivero) [reply src/classes/FORM_ServiceGiftEntry_TEST.cls]: 👍

**Pull request merged**